src/components/Sidebar.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Box,
+  CloseButton,
+  Flex,
+  useColorModeValue,
+  Drawer,
+  DrawerContent,
+  Text,
+  useDisclosure,
+  BoxProps,
+  Radio,
+  RadioGroup,
+  Button,
+} from "@chakra-ui/react";
+import { useHookstate } from "@hookstate/core";
+import React from "react";
+import store from "../store";
+import Goal from "../types/Goal";
+import Navbar from "./Navbar";
+
+export default function Sidebar() {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const [goals, setGoals] = React.useState<Array<Goal>>([]);
+  const [selectedGoal, setSelectedGoal] = React.useState<string>("");
+
+  const globalState = useHookstate(store);
+
+  React.useEffect(() => {
+    if (globalState.get().goals) {
+      setGoals(globalState.get().goals);
+    }
+  }, [globalState]);
+
+  return (
+    <Box minH="100vh" bg={useColorModeValue("gray.100", "gray.900")}>
+      <SidebarContent
+        onClose={() => onClose}
+        goals={goals}
+        selectedGoal={selectedGoal}
+        setSelectedGoal={setSelectedGoal}
+        display={{ base: "none", md: "block" }}
+      />
+      <Drawer
+        autoFocus={false}
+        isOpen={isOpen}
+        placement="left"
+        onClose={onClose}
+        returnFocusOnClose={false}
+        onOverlayClick={onClose}
+        size="full"
+      >
+        <DrawerContent>
+          <SidebarContent
+            onClose={onClose}
+            goals={goals}
+            selectedGoal={selectedGoal}
+            setSelectedGoal={setSelectedGoal}
+          />
+        </DrawerContent>
+      </Drawer>
+      <Navbar onOpen={onOpen} />
+      <Box ml={{ base: 0, md: 60 }} p="4">
+        <p>{selectedGoal ? "Selected goal" : "No goal selected"}</p>
+        <p>{goals.find((goal: Goal) => goal.id === selectedGoal)?.name}</p>
```

Comment:
This will probably get replaced down the line when we start working on the components for each goal, but it's a good base since we're able to display information specific to the goal which is selected